gdx/src/com/badlogic/gdx/graphics/g2d/DistanceFieldBitmapFont.java

```diff
@@ -0,0 +1,192 @@
+ * the names of its contributors may be used to endorse or promote products derived from this software without specific prior
+ * written permission.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,
+ * BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.badlogic.gdx.graphics.g2d;
+
+import com.badlogic.gdx.files.FileHandle;
+import com.badlogic.gdx.graphics.Texture.TextureFilter;
+import com.badlogic.gdx.graphics.glutils.ShaderProgram;
+import com.badlogic.gdx.utils.Array;
+
+/** Renders bitmap fonts using distance field textures, @see <a href="https://github.com/libgdx/libgdx/wiki/Distance-field-fonts">Distance Field Fonts Wiki Article</a>
+ * For usage init the SpriteBatch with the {@link #getDistanceFieldShader()} shader
+ * Attention: Breaks batching!
+ * @author Florian Falkner */
+public class DistanceFieldBitmapFont extends BitmapFont {
+	private static final String SMOOTHING_UNIFORM = "u_smoothing";
+	private float distanceFieldSmoothing;
+	private static ShaderProgram shader;
+	
+	public DistanceFieldBitmapFont (BitmapFontData data, Array<TextureRegion> pageRegions, boolean integer) {
+		super(data, pageRegions, integer);
+	}
+
+	public DistanceFieldBitmapFont (BitmapFontData data, TextureRegion region, boolean integer) {
+		super(data, region, integer);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile, boolean flip) {
+		super(fontFile, flip);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile, FileHandle imageFile, boolean flip, boolean integer) {
+		super(fontFile, imageFile, flip, integer);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile, FileHandle imageFile, boolean flip) {
+		super(fontFile, imageFile, flip);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile, TextureRegion region, boolean flip) {
+		super(fontFile, region, flip);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile, TextureRegion region) {
+		super(fontFile, region);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile) {
+		super(fontFile);
+	}
+
+	@Override
+	public BitmapFontCache newFontCache () {
+		for (int i = 0; i < regions.size; i++) {
+			getRegion(i).getTexture().setFilter(TextureFilter.Linear, TextureFilter.Linear);
+		}
```

Comment:
Texture configuration is separate from a font that uses the texture. You do it here so you don't have to do it elsewhere. If anything, it shouldn't happen every time a font cache is created since setting the texture filter binds the texture and makes GL calls. I guess the constructor is better.